schema.sql

```diff
@@ -6,119 +6,118 @@ CREATE DATABASE PiccadillyPizza;
 DROP TABLE IF EXISTS customer;
 CREATE TABLE customer (
   id SERIAL PRIMARY KEY,
-  name VARCHAR ( 255 ),
-  username VARCHAR ( 255 ) NULL,
-  password VARCHAR ( 255 ) NULL,
+  name VARCHAR ( 255 ) NOT NULL,
+  username VARCHAR ( 255 ) NOT NULL,
+  password VARCHAR ( 255 ) NOT NULL,
   phone_number INTEGER NULL
 );
 
 DROP TABLE IF EXISTS credit_card;
 CREATE TABLE credit_card (
   id SERIAL PRIMARY KEY,
-  customer_name VARCHAR ( 255 ) NULL,
-  card_number INTEGER NULL,
-  apartment_number VARCHAR ( 255 ),
-  street_address VARCHAR ( 255 ) NULL,
-  zipcode INTEGER,
-  customer_id INTEGER REFERENCES customer ( id ),
-  expiration_date TIMESTAMP NULL,
-  card_type VARCHAR ( 255 ),
-  city VARCHAR ( 255 ),
-  state VARCHAR ( 255 )
+  customer_name VARCHAR ( 255 ) NOT NULL,
+  card_number INTEGER NOT NULL,
+  billing_address VARCHAR ( 255 ) NOT NULL,
+  apartment_number VARCHAR ( 255 ) NULL,
+  zipcode INTEGER NOT NULL,
+  customer_id INTEGER REFERENCES customer ( id ) NOT NULL,
+  expiration_date TIMESTAMP NOT NULL,
+  card_type VARCHAR ( 255 ) NOT NULL,
+  city VARCHAR ( 255 ) NOT NULL,
+  state VARCHAR ( 255 ) NOT NULL
 );
 
 DROP TABLE IF EXISTS customer_address;
 CREATE TABLE customer_address (
   id SERIAL PRIMARY KEY,
-  customer_id INTEGER REFERENCES customer ( id ),
-  address_id INTEGER REFERENCES billing_address ( id )
+  customer_id INTEGER REFERENCES customer ( id ) NOT NULL,
+  address_id INTEGER REFERENCES billing_address ( id ) NOT NULL
 );
 
 DROP TABLE IF EXISTS purchase;
 CREATE TABLE purchase (
   id SERIAL PRIMARY KEY,
-  customer_id INTEGER REFERENCES customer ( id ),
-  completed BOOLEAN,
-  payment_method VARCHAR ( 255 ),
-  date_time TIMESTAMP,
-  total DECIMAL
+  customer_id INTEGER REFERENCES customer ( id ) NOT NULL,
+  completed BOOLEAN NOT NULL,
+  payment_method VARCHAR ( 255 ) NOT NULL,
+  date_time TIMESTAMP NOT NULL,
+  total DECIMAL NOT NULL
 );
 
 
 DROP TABLE IF EXISTS delivery;
 CREATE TABLE delivery (
   id SERIAL PRIMARY KEY,
-  status VARCHAR ( 255 ),
-  delivery_address INTEGER REFERENCES customer_address ( id ),
-  delivery_person VARCHAR ( 255 ),
-  -- consider delivery_person
-  purchase_id INTEGER REFERENCES purchase ( id ),
-  price DECIMAL,
-  tip DECIMAL
+  status VARCHAR ( 255 ) NOT NULL,
+  delivery_address INTEGER REFERENCES customer_address ( id ) NOT NULL,
+  delivery_person VARCHAR ( 255 ) NOT NULL,
```

Comment:
Feels like there's an opportunity for another relationship here - you are potentially repeating the same information over and over here.  Consider changing to a delivery_people table, and referencing the id as a foreign key in this table.